gradio/components/multimodal_textbox.py

```diff
@@ -0,0 +1,139 @@
+"""gr.MultimodalTextbox() component."""
+
+from __future__ import annotations
+
+from typing import Any, Callable, Literal
+
+from gradio_client.documentation import document
+
+from gradio.components.base import FormComponent
+from gradio.events import Events
+
+
+@document()
+class MultimodalTextbox(FormComponent):
+    """
+    Creates a textarea for users to enter string input or display string output and also allows for the uploading of multimedia files.
+
+    Demos: chatbot_multimodal
+    Guides: creating-a-chatbot
+    """
+
+    EVENTS = [
+        Events.change,
+        Events.input,
+        Events.select,
+        Events.submit,
+        Events.focus,
+        Events.blur,
+    ]
+
+    def __init__(
+        self,
+        value: list[dict[str, str]] | Callable | None = None,
```

Comment:
I think this would be a cleaner type for this object:
```
{
  "text": string,
  "files": list[string]
} 
```
either that, or a tuple:
```
Tuple[str, list[str]]
```

have a slight preference for the latter.

Reply:
That is definitely a simpler data structure, but I wanted it to be more aligned with chat + vision APIs like OpenAI's (https://platform.openai.com/docs/guides/vision). I'm not super tied to the current data structure tho, so open to changing it.

Reply:
I looked into this and personally what I would prefer is this format:

```py
{"text": ..., "files": [..., ...]}
```

For several reasons. 

(1) This is very similar to the `ImageEditor` format so we have a precedent.
(2) I originally thought it would make sense to align with `openai`'s format, but having looked at [their format](https://platform.openai.com/docs/guides/vision), they have specific keys like "image_url" etc. that we wouldn't want to bake into our component. So if a developer has to parse the data anyways, we should make it an easy format for them to parse
(3) Explicit keys are easier for developers to remember and for users reading their code than implicit tuple positions

E.g. its would be clear to a user of the code what this does:

import gradio as gr

def echo(x,y):
    x["text"]

gr.ChatInterface(echo, multimodal=True).launch()

Reply:
Makes sense, thanks for looking into this. making this change now

Reply:
Yeah. Following the discussions and after some thought I think what @abidlabs said above mid probably the best approach.